loopout.go

```diff
@@ -844,6 +850,23 @@ func (s *loopOutSwap) waitForHtlcSpendConfirmed(globalCtx context.Context,
 		// timer.
 		case notification := <-s.blockEpochChan:
 			s.height = notification.(int32)
+
+			maxPreimageRevealHeight := s.CltvExpiry -
+				MinLoopOutPreimageRevealDelta
+
+			// Check if the on-chain HTLC's CLTV has been reached.
+			// If so, we will time out the swap because it's no
+			// longer safe for us to reveal the preimage.
+			if s.height > maxPreimageRevealHeight {
```

Comment:
I think you also need to check whether we've revealed the preimage here? 

If we just give up when we reach our expiry height, we could land ourselves in a situation where we've revealed the preimage by broadcasting our sweep, but we haven't got our funds on-chain yet. The server _could_ try to claim off-chain and timeout on-chain, it won't, but we want to be trustless so we have to keep going for the on-chain sweep.

The real solution is to ensure that we sweep on-chain before the expiry at all costs - insert deadline aware sweeper here :D - but in the meantime we need to keep going for the sweep if we've revealed preimage.